File: src/main/java/com/reprezen/kaizen/oasparser/kmodel3/SecurityRequirement.java
package com.reprezen.kaizen.oasparser.kmodel3;

import com.reprezen.jsonoverlay.IJsonOverlay;
import com.reprezen.jsonoverlay.IModelPart;

import javax.annotation.Generated;
import java.util.Map;

public interface SecurityRequirement extends IJsonOverlay<SecurityRequirement>, IModelPart<OpenApi3, SecurityRequirement> {

    // Requirement
    @Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
    Map<String, SecurityParameter> getRequirements();

    @Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
    Map<String, SecurityParameter> getRequirements(boolean elaborate);

    @Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
    boolean hasRequirements();

    @Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
    boolean hasRequirement(String name);

    @Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
    SecurityParameter getRequirement(String name);

    @Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
    void setRequirements(Map<String, SecurityParameter> requirements);

    @Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
    void setRequirement(String name, SecurityParameter requirement);

    @Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
    void removeRequirement(String name);
}

File: parser/src/main/kotlin/com/reprezen/kaizen/oasparser/model3/SecurityRequirement.kt
package com.reprezen.kaizen.oasparser.model3

import com.reprezen.jsonoverlay.parser.Generated
import com.reprezen.jsonoverlay.IJsonOverlay
import com.reprezen.jsonoverlay.IModelPart
import com.reprezen.kaizen.oasparser.model3.SecurityParameter
import kotlin.collections.Map

interface SecurityRequirement : IJsonOverlay<SecurityRequirement>, IModelPart<OpenApi3, SecurityRequirement> {

	// Requirement
	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getRequirements() : MutableMap<String, SecurityParameter>

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun hasRequirements() : Boolean

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun hasRequirement(name : String) : Boolean

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getRequirement(name : String) : SecurityParameter?

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun setRequirements(requirements : MutableMap<String, SecurityParameter>)

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun setRequirement(name : String,requirement : SecurityParameter)

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun removeRequirement(name : String)

}
